Doc-3593 screenshot hygiene (#2957)
harness/developer-hub: docs/platform/8_Pipelines/w_pipeline-steps-reference/yaml-reference-cd-pipeline.md

New to Harness YAML? See [Harness YAML Quickstart](../harness-yaml-quickstart.md).

## Viewing the YAML Schema

> The Harness YAML schema is over 20,000 lines long, and we are investigating how it can be exposed in a way that is easy to navigate.

Reviewer: ```suggestion
The Harness YAML schema is available at [github.com/harness/harness-schema](https://github.com/harness/harness-schema). This repository contains the schema definition for Harness pipeline configuration files, enabling users to define and customize their pipelines effortlessly. This repo provides steps on how to embed the YAML schema in Visual Studio Code and Sublime Text.
```

Reviewer: honestly, this entire topic could be revised to give a quick summary and then just point to the YAML schema repo.

> In the meantime, you can use [Chrome DevTools](https://developer.chrome.com/docs/devtools/) to view the schema.

Reviewer: ```suggestion
In addition to the YAML schema is available at [github.com/harness/harness-schema](https://github.com/harness/harness-schema), you can use [Chrome DevTools](https://developer.chrome.com/docs/devtools/) to view the schema.
```

1. Open Chrome DevTools and select the **Network** tab.
2. Select the `yaml-schema` item.
3. Select the **Preview** tab.
4. Find the `definitions` section.

<!-- ![](./static/yaml-reference-cd-pipeline-00.png) -->

<docimage path={require('./static/yaml-reference-cd-pipeline-00.png')} />

## Pipeline Studio YAML Editor

The Harness Pipeline Studio has both a Visual Editor and a YAML Editor.

> The best way to get started with Harness YAML is to create a pipeline in the Visual Editor and then switch to the YAML Editor. If you haven't created a pipeline yet, try a [tutorial](/docs/getting-started/quickstarts/) to get started with Harness.

Reviewer: ```suggestion
The best way to get started with Harness YAML is to create a pipeline in the Visual Editor and then switch to the YAML Editor. If you haven't created a pipeline yet, try a [tutorial](https://developer.harness.io/tutorials) to get started with Harness.
```

For example, here is a Connector with the name `GCP Example` and Id `GCP_Example`. You can see the Id used in the YAML as `connectorRef: GCP_Example`:

| Connector | YAML |
| --------- | ---- |
| ![](.static/../static/yamlrefcdpipeline1.png) | ![](./static/yamlrefcdpipeline2.png) |

```yaml
...
type: Gcr
spec:
  connectorRef: GCP_Example
  imagePath: library/bar
  registryHostname: gcr.io
  tag: <+input>
identifier: foo
...
```

Once you have entered the Id in `connectorRef`, you can use autocomplete to view the remaining settings:

![](./static/yaml-reference-cd-pipeline-03.png)

## Schema Overview

> Harness Pipeline YAML lets you model your release process declaratively. Each Pipeline entity, component, and setting has a YAML entry.

Reviewer: ```suggestion
Harness pipeline YAML lets you model your release process declaratively. Each pipeline entity, component, and setting has a YAML entry.
```

        delegateSelectors:  
            - example
```

> You reference a Connector in your Pipeline by using its Id in `connectorRef`:

Reviewer: ```suggestion
You reference a connector in your pipeline by using its Id in `connectorRef`:
```
